an-rope/src/lib.rs

```diff
@@ -424,25 +425,26 @@ impl Rope {
     /// ```
     /// use an_rope::Rope;
     /// let mut an_rope = Rope::from(String::from("efgh"));
-    /// an_rope.prepend(Rope::from(String::from("abcd")))
-    /// assert_eq!(an_rope, "abcdefgh");
+    /// an_rope.prepend(Rope::from(String::from("abcd")));
+    /// assert_eq!(&an_rope, "abcdefgh");
     /// ```
     /// ```
     /// use an_rope::Rope;
     /// let mut an_rope = Rope::from("");
     /// an_rope.prepend(Rope::from("abcd"));
-    /// assert_eq!(an_rope, "abcd");
+    /// assert_eq!(&an_rope, "abcd");
     /// ```
     /// ```
     /// use an_rope::Rope;
     /// let mut an_rope = Rope::from("abcd");
     /// an_rope.prepend(Rope::from(""));
-    /// assert_eq!(an_rope, "abcd");
+    /// assert_eq!(&an_rope, "abcd");
```

Comment:
As discussed, let's move the no-op tests to unit tests and make the doc tests illustrate non-trivial usage.